src/esn.calendar.libs/app/services/calendar-api.js

```diff
@@ -316,5 +317,21 @@ require('./http-response-handler.js');
       return calDavRequest('get', calendarPath, { Accept: CAL_ACCEPT_EXPORT_HEADER })
         .then(calHttpResponseHandler(200, _.property('data')));
     }
+
+    /**
+     * POST resuest used to generate token for secret link to download ics calendar
+     * @param {Object} jwtPayload   paylod to generate token
+     */
+    function generateToken(jwtPayload) {
+
+      return calendarRestangular.one('calendars/generateJWTforSecretLink').customPOST(jwtPayload)
+        .then(calHttpResponseHandler(200, _.property('data')))
+        .catch(function(error) {
+          notificationFactory.weakError('Failed to generate token for secret', 'Cannot join the server, please try later');
```

Comment:
```suggestion
          notificationFactory.weakError('Failed to generate token for the secret address', 'Cannot join the server, please try later');
```

Just `for secret` is too ambiguous for users.

Reply:
Also no translation.